common/log/tag/tags.go

```diff
@@ -630,6 +630,11 @@ func Attempt(attempt int32) ZapTag {
 	return NewInt32("attempt", attempt)
 }
 
+// AttemptWithUnexpectedError returns tag for AttemptWithUnexpectedError
+func AttemptWithUnexpectedError(attempt int) ZapTag {
+	return NewInt("attempt-with-unexpected-error", attempt)
```

Comment:
nit: make the type consistent with Attempt(). I am fine with either Int or Int32.